packages/@ember/octane-app-blueprint/index.js

```diff
@@ -11,30 +11,27 @@ module.exports = {
   filesToRemove: [],
 
   locals(options) {
-    let name = dasherize(options.entity.name);
-    let blueprintVersion = require('./package').version;
 
-    let entity = options.entity;
-    let rawName = entity.name;
-    let namespace = classify(rawName);
+    return getURLFor('canary').then(url => {
+      let emberCanaryVersion = url;
+      let name = stringUtil.dasherize(options.entity.name);
+      let entity = options.entity;
+      let rawName = entity.name;
+      let namespace = stringUtil.classify(rawName);
 
-    return {
-      name,
-      modulePrefix: name,
-      namespace,
-      yarn: true,
-      welcome: false,
-      blueprintVersion,
-      versions: {
-        emberCLI: 'github:ember-cli/ember-cli#31ed51040c51e6d47c9fc3bb860f46108feefea2',
-        ember: this.emberCanaryUrlForVersion('c24bc23e4139c90c8d8d96c4234d9c0c19e5c594'),
-        emberData: '~3.7.0',
-      }
-    };
-  },
-
-  emberCanaryUrlForVersion(version) {
-    return `https://s3.amazonaws.com/builds.emberjs.com/canary/shas/${version}.tgz`
+      return {
+        name,
+        modulePrefix: name,
+        namespace,
+        yarn: options.yarn,
+        welcome: options.welcome,
+        emberCanaryVersion,
+        versions: {
+          emberCLI: 'github:ember-cli/ember-cli',
+          emberData: '~3.7.0',
```

Comment:
if we're gonna get latest canary url, maybe we get the latest published ember-data?
and also get the latest ember-cli commit (at present, it'll default to master, which'll result in some caching issues that using the sha helps get around)

Reply:
@NullVoxPopuli Do you know how to get them, I could not figure out when reading`ember-source-channel-url` 

https://github.com/ember-cli/ember-source-channel-url/blob/master/src/index.js

I also thought about that, but this may be a minor issue or nice to have. With every `ember-cli` upgrade, the blueprints/package.json are manually updated. Atm, we will have to do the same.

Reply:
A possible solution would be to add `ember-data` and `ember-cli` with the `afterInstall/this.addAddonsToProject` and will pick up the latest releases. 

Thoughts?

Reply:
yeah, it's pretty easy:

npm packages: https://www.npmjs.com/package/latest-version
git revision: 

```
 curl https://api.github.com/repos/ember-cli/ember-cli/commits/master -H "Accept: application/vnd.github.VERSION.sha"
de27b18941f6ef80cc9185dc18ad7eec60a5d994
```

Reply:
I think the addAddons / afterInstall stuff still just uses the selected package manager